Sources/NIOHTTP1/HTTPServerUpgradeHandler.swift

```diff
@@ -235,17 +235,22 @@ public final class HTTPServerUpgradeHandler: ChannelInboundHandler, RemovableCha
                 self.removeExtraHandlers(context: context).flatMap {
                     self.sendUpgradeResponse(context: context, upgradeRequest: request, responseHeaders: finalResponseHeaders)
                 }.flatMap {
-                    self.removeHandler(context: context, handler: self.httpEncoder)
-                }.map {
+                    context.pipeline.removeHandler(self.httpEncoder)
+                }.flatMap { () -> EventLoopFuture<Void> in
```

Comment:
The semantics here are slightly awkward: 
- `self.upgradeCompletionHandler(_:)` can fire followed by
- `upgrader.upgrade(context:upgradeRequest:)` which can fail

I think this is okay though: `upgradeCompletionHandler` is documented as being called after the HTTP upgrade is complete (which it is: the upgrade response was successfully sent.) This is distinct from the pipeline upgrade completing which is signalled by `HTTPServerUpgradeEvents.upgradeComplete` so I think this is fine, if a little unclear.